### DISCUSSION TITLE
Code Coverage Not Working

### DISCUSSION THREAD

**Discussion by TheFlyer1983** [vitest-dev/vitest#1838] [Q&A]
Before I start, I understand that code coverage is handled by c8, but I thought that there might be something I'm doing wrong in my config.

I am running Vue 3.2.25, Vite 3.0.5, Vitest 0.21.1
node v16.15.1

I have 118 tests in my project, and all of them pass, but my coverage is doesn't show at all. I know questions have been asked like this before, but I've not found the answer.

![image](https://user-images.githubusercontent.com/36920884/184124495-321d610e-ada7-4886-bb42-5ce5c4682ff9.png)

I have this in my `vite.config.js`:
![image](https://user-images.githubusercontent.com/36920884/184124659-a827cb2b-b05e-410e-a061-b7f971499bfe.png)

If I uncomment the `//all`, then it just shows all my files with 0 lines covered:
![image](https://user-images.githubusercontent.com/36920884/184124891-d7d79bb0-07fb-48e1-8536-6d0e1859aba5.png)

I know I'll probably need to ask in the c8 repo, but any idea's where I'm going wrong?

**Comment by TheFlyer1983** [comment]
This was due to other areas of my vite.config.ts, and is now resolved. This can be closed.

**Reply by sheremet-va** [reply]
It might be helpful to others finding this question if you will specify how you fixed it. Thank you 😄

**Comment by TheFlyer1983** [comment]
No worries,

So I had this config:
```
export default ({ mode }) => {
  process.env = { ...process.env, ...loadEnv(mode, process.cwd()) };

  return defineConfig({
    define: {
      'process.env': { ...process.env, ...loadEnv(mode, process.cwd()) }
    },
    ... other config,
    test: {
      globals: true,
      reporters: ['json', 'verbose'],
      outputFile: './tests/test-report.json',
      environment: 'jsdom',
      setupFiles: './tests/setup.js',
      coverage: {
        reporter: ['text', 'json', 'html'],
        exclude: [
          'node_modules/**',
          'coverage',
          'public/**',
          '**/*{.,-}spec.ts',
          '**/vite.config.ts',
          '**/src/*.d.ts',
          '**/src/main.ts'
        ],
        extension: ['.ts', '.vue']
      }
    }
  });
};
```

I can't remember when I needed the process.env bit at the top, and if I removed this and the `define` node, and had my `defineConfig` as the `export default` then the coverage seemed to work.

```
export default defineConfig({
  server: {
    port: 8080
  },
  ... other config,
  test: {
      globals: true,
      reporters: ['json', 'verbose'],
      outputFile: './tests/test-report.json',
      environment: 'jsdom',
      setupFiles: './tests/setup.js',
      coverage: {
        reporter: ['text', 'json', 'html'],
        exclude: [
          'node_modules/**',
          'coverage',
          'public/**',
          '**/*{.,-}spec.ts',
          '**/vite.config.ts',
          '**/src/*.d.ts',
          '**/src/main.ts'
        ],
        extension: ['.ts', '.vue']
     }
  }
}
```

**Reply by AlexMarlow25** [reply]
I have the same issue with my `--coverage`, but after I removed `process.env = { ...process.env, ...loadEnv(mode...` stuff from config everything works fine! So not to break my dev and build scripts I added check on `mode !== 'test'` in my `vite.config`, like this:
```
if (mode !== 'test') {
    process.env = { ...process.env, ...loadEnv(mode, appRoot) };
}
```

Thank you a lot for that hint!

**Comment by mpaccione** [comment]
Hi I have the issue where no matter what I do no coverage is generated. I too was doing a similar approach with my process env. However when making the changes listed above there is no change in my outcome.

EDIT
Need this in your test.coverage block
        reportOnFailure: true,
Why its not default to true surely costs some engineers premature hair loss.

**Reply by ThangZaThangWoodside** [reply]
Thanks, this was it :)
